Makefile.am

```diff
@@ -1275,6 +1278,8 @@ imap_httpd_SOURCES = \
 	imap/http_ws.h \
 	imap/httpd.c \
 	imap/httpd.h \
+	imap/itip_support.c \
+	imap/itip_support.h \
 	imap/jcal.c \
```

Comment:
Since itip_support.c and caldav_util..c are used from lmtpd and httpd, it makes sense to move both files into libcyrus_imap.  It is a library for shared code.

Reply:
The problem with that is that there are 2 global variables (httpd_userid and namespace_calendar) that would have to defined by every binary that links libcyrus_imap.  I'd rather not have to do that.

Reply:
I guess its possible to include this everywhere, but do IMAP, and POP really care about iTIP?

Reply:
The overall shared code library handling is a bit random.  E.g. spool.c is included in libcyrus_imap and as source file in httpd and others.  I think the overall criterion to include something in a shared library is, if a source file is needed by more than one executable. The httpd_userid and imapd_userid global variables serve the same purpose, so they can share also the same name.

I think the ultimate goal shall be to move all shared code in at most two libraries.  As far as I remember I implemented autocreate (of the imap and sieve user) on the first CalDAV login and this made libcyrus_imap depend on libcyrus_sieve and libcyrus_sieve depend on libcyrus_imap…

imap/zoneinfo_db.c is another candidate to put in a shared library.

As a rule of thumb global symbols are not good.  If the fatal() implementation can be put (as WEAK symbol in some shared library, them most of the fatal() implementations could be removed.

Reply:
@brong Are you still wanting code changes?  Are you happy with the tests in https://github.com/cyrusimap/cassandane/pull/167?

Reply:
Have replied on that - the tests look OK.  Gonna read the code for careful now.